mysql/datadog_checks/mysql/statement_samples.py

```diff
@@ -725,13 +765,14 @@ def _explain_statement(self, cursor, statement, schema, obfuscated_statement):
                         (time.time() - start_time) * 1000,
                         tags=self._tags + ["strategy:{}".format(strategy)],
                     )
-                    return plan
+                    return plan, None, None
             except pymysql.err.DatabaseError as e:
                 if len(e.args) != 2:
                     raise
                 # we don't cache failed plan collection failures for specific queries because some queries in a schema
                 # can fail while others succeed. The failed collection will be cached for the specific query
                 # so we won't try to explain it again for the cache duration there.
+                db_explain_error, err = DBExplainError.database_error, e
```

Comment:
I wonder if we should collect all of the errors, by strategy? The problem with this approach is that we'll only get the error for the last strategy, potentially hiding other setup issues. 

Perhaps for mysql instead of just one `(code, reason)` we have a map of `strategy -> (code, reason)`? Maybe under a different key: `collection_error_by_strategy`

@justiniso what do you think?

Reply:
Discussed offline, but for context, there are cases where we wont have a strategy to make a mapping of `strategy -> (code, msg)` (i.e `_can_explain()`) so we were thinking, what if we simplify it to where we structure it as`[{strategy, code, message}, ...]` under `collection_errors` to be more universal?

Reply:
Decided to implement this approach, we can revert if necessary.